### PR TITLE
Tighten exercises for classes

### PR DESCRIPTION
**Pull request by njpearman**
This PR tightens up the setup for the classes exercises, following on from #3. It moves the side nav to the RHS, as I think this is less distracting. I've also adjusted some of the text.

I have also installed a `babel` plug-in (`@babel/plugin-proposal-class-properties`) so the that I can include an exercise about private class properties. This has resulted in an update to the lock files. However, I have done this using an older version of `npm` so that the `lockfileVersion` is still `1`, and the plug-in is set at the same version as `@babel/preset-env` (`7.3.4`).

### CODE REVIEW

**Comment by njpearman** [comment]
Uff, bit of a GH cli fail as I've opened this against `master` rather than the branch for #3... I'll just use this PR in its entirety and close #3.

**Comment by njpearman** [comment]
I should have added that for this PR I'd just like a technical review of the setup for this code, not a review of the exercises themselves. This will be done separately.

As such, it just makes #3 more complete.

File: .babelrc [hunk 1]
```diff
@@ -7,5 +7,6 @@
         }
       }
     ]
-  ]
+  ],
+  "plugins": ["@babel/plugin-proposal-class-properties"]
```
Comment (hcarver) [inline .babelrc:11]: Maybe obvious, but I wanted to say it anyway. If this is still a proposal, we should emphasise that to the learner in the slides that relate.

File: src/classes/classes.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
+const maxItemsToDisplay = 5;
+const CACHE = {};
```
Comment (hcarver) [inline src/classes/classes.js:2]: Not critical, but there are load of other things that could be `const` here, if you wanted.

File: src/classes/classes.js [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+// data feed setup
+let dataFeeds = {
+  reddit: {
+    id: "reddit",
+    url:
+      "https://api.reddit.com/api/subreddit_autocomplete_v2.json?limit=10&include_over_18=false&query=",
+    normaliseData: function(data) {
+      let normalisedData = [];
+      var data = data.data;
```
Comment (hcarver) [inline src/classes/classes.js]:
```suggestion
    normaliseData: function(data) {
      let normalisedData = [];
      data = data.data;
```

☝️ I think this should be OK, and avoids the ugly `var`.

File: src/index.html [hunk 4]
```diff
@@ -15,49 +15,57 @@
...
         </div>
+        <div class="row">
+          <div id="last-search-time" class="col-sm-12"></div>
+        </div>
       </div>
-      <div class="row">
-        <div id="last-search-time" class="col-sm-12"></div>
+      <div class="sidebar" style="flex: 1">
+        <div>
+          <em>Module specific exercises:</em>
```
Comment (hcarver) [inline src/index.html]:
```suggestion
          <em>Module-specific exercises:</em>
```

File: src/classes/classes.js [hunk 5]
```diff
@@ -7,7 +7,11 @@ function updateCache(id, searchTerm, data) {
 }
 
 function itemFromCache(id, searchTerm) {
-  CACHE[id] && CACHE[id][searchTerm];
+  if (CACHE[id] && CACHE[id][searchTerm]) {
+    return CACHE[id][searchTerm];
+  }
+
+  return null
```
Comment (hcarver) [inline src/classes/classes.js]: The commit title is to "Complete" this function. I want to highlight that the previous version was also complete and functionally equivalent to this version.

**Review by hcarver — APPROVED** [review]
Thanks for avoiding the npm version upgrade. A few comments, but I think only one (the suggestion) that's necessary.

File: .babelrc (hunk 1 above)
Comment (njpearman) [reply .babelrc:11]: Private props are no longer a proposal, I think: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields#specifications. I don't really know the JS spec process though, so I might be wrong.

Also, I think needing this babel plugin is a result of using the older version of `preset-env` / `npm`. The plugin is currently listed as included in `preset-env` (see https://babeljs.io/docs/en/babel-plugin-proposal-class-properties), but I needed to add the plugin to get private props to work in this setup.

File: src/classes/classes.js (hunk 2 above)
Comment (njpearman) [reply src/classes/classes.js:2]: Yep, I'll look at tidying up the overall codebase here as a follow-up.

File: src/classes/classes.js (hunk 5 above)
Comment (njpearman) [reply src/classes/classes.js]: Yeah, I agree with your comment and expected the same, but the functionality didn't seem to work with the more compact version... I'll go back and look again.

Comment (njpearman) [reply src/classes/classes.js]: I've added an explicit return to the original version, and it works fine. I must've been confused at the point where I decided to add this long-form conditional.

Comment (hcarver) [reply src/classes/classes.js]: Ah yes, you're right. The original needed a `return` or to be written as a single-expression arrow function, so it was broken as it was.

**Pull request merged**